pdcupdater/handlers/modules.py

```diff
@@ -0,0 +1,267 @@
+            return False
+
+        if state not in self.relevant_states:
+            log.debug("Non-relevant module state '{}', skipping.".format(
+                state))
+            return False
+
+        return True
+
+    def handle(self, pdc, msg):
+        body = msg['msg']
+        state = body['state']
+
+        if state not in self.relevant_states:
+            log.warn("Non-relevant module state '{}', skipping.".format(
+                state))
+            return
+
+        unreleased_variant = self.get_or_create_unreleased_variant(pdc, body)
+
+        if 'topdir' in body:
+            self.handle_new_tree(pdc, body, unreleased_variant)
+
+    def get_mmd_from_scm(self, scmurl):
+        with TmpDir(prefix="pdcupdater-") as tmpdir, PushPopD(tmpdir), \
+                open(os.devnull, "w") as devnull:
+            m = self.scmurl_re.match(scmurl)
+            if not m:
+                log.error("Can't parse SCM URL: {}".format(scmurl))
+                raise RuntimeError("Can't parse SCM URL: {}".format(scmurl))
+            giturl = m.group('giturl')
+            repopath = m.group('repopath')
+            modpath = m.group('modpath')
+            revision = m.group('revision')
+            while repopath.endswith("/"):
+                repopath = repopath[:-1]
+            modname = repopath.rsplit("/", 1)[1]
+            if modname.endswith(".git"):
+                modname = modname[:-4]
+
+            log.debug("Cloning {}".format(giturl))
+            check_call(["git", "clone", "-n", giturl, modname],
+                       stdout=devnull, stderr=STDOUT)
+            os.chdir(modname)
+
+            log.debug("Checking out revision {}".format(revision))
+            check_call(["git", "reset", "--hard", revision],
+                       stdout=devnull, stderr=STDOUT)
+
+            mmd_yaml = modname + ".yaml"
+            if modpath:
+                mmd_yaml = "/".join((modpath, mmd_yaml))
+            log.debug("Reading/parsing {}".format(mmd_yaml))
+            mmd = modulemd.ModuleMetadata()
+            mmd.load(mmd_yaml)
+
+            return mmd
+
+    def create_unreleased_variant(self, pdc, body):
+        """Creates an UnreleasedVariant for a module in PDC. Checks out the
+        module metadata from the supplied SCM repository (currently only
+        anonymous GIT is supported)."""
+
+        scmurl = body['scmurl']
+        mmd = self.get_mmd_from_scm(scmurl)
+
+        runtime_deps = []
+        for dep, ver in mmd.requires.items():
+            if ver is not None:
+                runtime_deps.append("{} >= {}".format(dep, ver))
+            else:
+                runtime_deps.append(dep)
+
+        build_deps = []
+        for dep, ver in mmd.buildrequires.items():
+            if ver is not None:
+                build_deps.append("{} >= {}".format(dep, ver))
+            else:
+                build_deps.append(dep)
+
+        koji_tag = body['koji_tag']
```

Comment:
And, right now I assume that `module_id` or `module_uid` contain name, version, release. Right or wrong?